### DISCUSSION TITLE
KafkaBrokerLoggingConfigurationDiff - Invalid logging level

### DISCUSSION THREAD

**Discussion by armisz** [strimzi/strimzi-kafka-operator#4326] [Q&A]
Hello,

I have an issue setting the log configuration for the Cluster Operator on Strimzi 0.21.0 on OpenShift 3.11. Maybe it's a bug, maybe it's me, I have double checked everything but can't see any problem with my setup.

The error message is flooding my log on every reconciliation:
`2021-01-27 15:29:48 WARN  KafkaBrokerLoggingConfigurationDiff:252 - Invalid logging level: . Using WARN as a failover.`

My log configuration is this:
```
sh-4.2$ cat custom-config/log4j2.properties 
name = COConfig
monitorInterval = 30

appender.console.type = Console
appender.console.name = STDOUT
appender.console.layout.type = PatternLayout
appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss} %-5p %c{1}:%L - %m%n

rootLogger.level = INFO
rootLogger.appenderRefs = stdout
rootLogger.appenderRef.console.ref = STDOUT
rootLogger.additivity = false

# Kafka AdminClient logging is a bit noisy at INFO level
logger.kafka.name = org.apache.kafka
logger.kafka.level = WARN
logger.kafka.additivity = false

# Zookeeper is very verbose even on INFO level -> We set it to WARN by default
logger.zookeepertrustmanager.name = org.apache.zookeeper
logger.zookeepertrustmanager.level = WARN
logger.zookeepertrustmanager.additivity = false

# Keeps separate level for Netty logging -> to not be changed by the root logger
logger.netty.name = io.netty
logger.netty.level = INFO
logger.netty.additivity = false
```

All log levels look good for me. Any hint is very much appreciated.

It would be nice if there was a hint to the location of the problem log level in future.

Thanks in advance
Michael

**Comment by scholzj** [comment]
That looks like the log4j2 properties for the Cluster Operator. Are you trying to use it with Kafka broker? Kafka is using Log4j 1 => so this would not work. Or what exactly are you trying to do?

**Reply by armisz** [reply]
It is correct, those are log4j2 properties for the Cluster Operator, mounted through a ConfigMap. The error message in my original post is taken from the Cluster Operator log.

I do not try to set those log properties for the Kafka broker. For Kafka an ZK I have dedicated log4j properties set through ConfigSets.

**Reply by scholzj** [reply]
Ok ... so can you share the:
* Full Cluster Operator log with the error
* The related custom resources you use
* The deployment of the Cluster Operator?

**Reply by armisz** [reply]
Here are the files you requested:

[strimzi-cluster-operator-config.yml.txt](https://github.com/strimzi/strimzi-kafka-operator/files/5886377/strimzi-cluster-operator-config.yml.txt)
[strimzi-cluster-operator-deployment.yml.txt](https://github.com/strimzi/strimzi-kafka-operator/files/5886378/strimzi-cluster-operator-deployment.yml.txt)
[cluster-operator.log](https://github.com/strimzi/strimzi-kafka-operator/files/5886379/cluster-operator.log)

**Reply by scholzj** [reply]
Can you also share the custom resources you deployed? From the operator log it should be Kafka CR called `abc-strimzi-kafka` in namespace `abc-test`. As I said, I do not think this is related to your cluster operator logging but to your Kafka cluster.

**Reply by armisz** [reply]
[kafka-logging.yml.txt](https://github.com/strimzi/strimzi-kafka-operator/files/5887200/kafka-logging.yml.txt)
[kafka.yml.txt](https://github.com/strimzi/strimzi-kafka-operator/files/5887202/kafka.yml.txt)

I have expected that the log4j2 properties have only an effect on the Cluster Operator and not on the other managed Kafka and ZK instances.

**Reply by scholzj** [reply]
I do not think this is related to the `log4j2.proepties` file. It should be all about our `log4j.proeprties` file for the KAfka cluster. In particular it does not like the `log4j.rootLogger=${kafka.root.logger.level}, CONSOLE` from it. If you replace it with `log4j.rootLogger=INFO, CONSOLE` directly instead of using the variable the error should disappear.

@sknot-rh Is this expected? Can this be somehow fixed or improved?

**Reply by sknot-rh** [reply]
Let me investigate.

**Reply by sknot-rh** [reply]
Ok, so it is a bug. The configuration uses the `kafka.root.logger.level` before it is defined. Fixed in https://github.com/strimzi/strimzi-kafka-operator/pull/4340
